src/angular-signature-pad.js

```diff
@@ -60,8 +64,19 @@ function BmSignaturePadController ($scope, $element, $attrs, $window, $log) {
     if (vm.resize) {
       vm.resize({
         $fn: () => {
-          signaturePad.resize(element.clientWidth, element.clientHeight, vm.scaleDown && vm.scaleDown())
-          vm.ngModel.$setViewValue(getSignature())
+          const width = element.clientWidth
+          const height = element.clientHeight
+          // If canvas is empty we dont need to call resize or set the model
+          if (vm.ngModel.$isEmpty()) {
+            canvas.width = width
+            canvas.height = height
+          } else {
+            const contentPreserved = canvasManipulation.resize(canvas, width, height, vm.scaleDown && vm.scaleDown())
```

Comment:
just thinking out loud here : I've been thinking about the image fidelity loss when the image is resized often that you mentioned. Will this resize function be run when the user rotates the phone? I'm wondering if the form has a few camera fields, so the user is constantly rotating the phone taking photos , fill out form, taking photos, would the degradation of the signature be much? or was it just resizing browser windows and the huge amount of resize events that throws that caused the image to get blurrier ?